### ISSUE TITLE
Owned entities whose parent entity's ID column name is not standard

### ISSUE THREAD

**Issue by TimmVonPuttkamer** [efcore/EFCore.NamingConventions#36]
I came across the following problem when using owned entities together with UseSnakeCaseNamingConvention. The minimal example works as follows:

```csharp
[Owned]
public class OwnedEntity
{
    public string SomeString { get; set; }
}

public class Parent
{
    // Works, if we use the standard naming convention for primary keys, i.e.
    // [Key]
    // public int ParentId { get; set; }
    // 

    // The following leads to an error:
    [Key]
    public int SomeDifferentlyNamedParentId { get; set; }

    public OwnedEntity Owned { get; set; }
}

public class DatabaseContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        => optionsBuilder
            .UseNpgsql("...")
            .UseSnakeCaseNamingConvention();

    public DbSet<Parent> Parents { get; set; }
}
```

When running ```bash dotnet ef migrations add InitialCreate```, I get the following error:
```
System.InvalidOperationException: The keys {'SomeDifferentlyNamedParentId'} on 'Parent' and {'ParentSomeDifferentlyNamedParentId'} on 'OwnedEntity' are both mapped to 'parents.pk_parents' but with different columns ({'some_differently_named_paren
t_id'} and {'parent_some_differently_named_parent_id'}).
   at Microsoft.EntityFrameworkCore.Infrastructure.RelationalModelValidator.ValidateSharedKeysCompatibility(IReadOnlyList`1 mappedTypes, String tableName, IDiagnosticsLogger`1 logger) ...
```
If I try to generate the same migration but without UseSnakeCaseNamingConvention, it works. Is there something I am missing?

I am using EF Core 3.1.7 and EFCore.NamingConventions 1.1.0

**Comment by roji** [comment]
Thanks, I can confirm this still happens with the recently-released 5.0.0-preview8, will try to fix this for rc1.

**Comment by roji** [comment]
Confirmed that this is gone in 5.0.0-rc1.
